test/e2e/nodenetworkpolicy_test.go

```diff
@@ -891,3 +894,47 @@ func testNodeACNPNestedIPBlockClusterGroupCreateAndUpdate(t *testing.T) {
 	}
 	executeTests(t, testCase)
 }
+
+func testNodeACNPAuditLogging(t *testing.T, data *TestData) {
+	// In a Kind cluster, each Kubernetes Node runs as a Docker container. The iptables LOG target doesn't generate
+	// kernel logs inside containers unless /proc/sys/net/netfilter/nf_log_all_netns is set to 1.  However, this setting
+	// cannot be modified in GitHub Actions.
+	skipIfProviderIs(t, "kind", "The iptables LOG action doesn't generate log in container")
+	randomLogLabel := uuid.New().String()[:12] // Generate a random logLabel with 12 characters long.
+	toBeTruncated := "thisPartShouldBeTruncated"
+	logLabel := randomLogLabel + toBeTruncated // Append the suffix to the logLabel.
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("acnp-drop-x-to-y-egress").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{NodeSelector: map[string]string{labelNodeHostname: nodes["x"]}}})
+	builder.AddEgress(ProtocolTCP, &p80, nil, nil, nil, nil, nil, nil, nil, nil, map[string]string{labelNodeHostname: nodes["y"]}, nil,
+		nil, nil, nil, nil, nil, crdv1beta1.RuleActionDrop, "", "", nil)
+	builder.AddEgressLogging(logLabel)
+
+	acnp, err := k8sUtils.CreateOrUpdateACNP(builder.Get())
+	failOnError(err, t)
+	failOnError(data.waitForACNPRealized(t, acnp.Name, policyRealizedTimeout), t)
+
+	expectedLogPrefix := fmt.Sprintf("Antrea:O:Drop:%s:", randomLogLabel)
+
+	// Before generating some traffic, there should be no corresponding kernel log on hostNetwork Pod x/a, and the kernel
+	// logs should not contain the generated random logLabel.
+	_, stdout, stderr, err := data.RunCommandOnNode(nodes["x"], "dmesg")
+	if err != nil || stderr != "" {
+		t.Fatalf("Got error: %v, %v", err, stderr)
+	}
+	require.NotContains(t, stdout, expectedLogPrefix)
+
+	// Generate some traffic that will be dropped by acnp-drop-x-to-y-egress.
+	_, err = k8sUtils.Probe(getNS("x"), "a", getNS("y"), "a", p80, ProtocolTCP, nil, nil)
+	require.NoError(t, err)
+
+	// After generating some traffic, there should be corresponding kernel logs on hostNetwork Pod x/a, and the kernel
+	// logs should contain the generated random logLabel and not contain the part that should be truncated.
+	_, stdout, stderr, err = data.RunCommandOnNode(nodes["x"], "dmesg")
+	if err != nil || stderr != "" {
+		t.Fatalf("Got error: %v, %v", err, stderr)
+	}
+	require.Contains(t, stdout, expectedLogPrefix)
+	require.NotContains(t, stdout, toBeTruncated)
```

Comment:
nit: `assert` is more appropriate than `require` here IMO

IMO we should do something better here. We should locate the actual log entry (maybe with a regex) and validate independent fields.

BTW, does the log always show up immediately or should we use `Poll` to avoid flakes?

Reply:
I have updated this part:
- Add a regex to match source/destination IP, random log label
- Use `assert.EventuallyWithT` to avoid flakes.